service/utils.go

```diff
@@ -162,6 +162,7 @@ func versionResource(apiContext *api.ApiContext, catalogName string, template mo
 		if showUpgradeVersion(version, upgradeVersion, rancherVersion) {
 			route := generateVersionId(catalogName, template, upgradeVersion)
 			link := apiContext.UrlBuilder.ReferenceByIdLink("template", route)
+			link = link + "?projectId=" + envId
```

Comment:
Not safe to modify the link by just adding `?projectId`. If something in that function added a query parameter, you'd be in trouble.
Probably need to use a function on the URL struct to add a url parameter properly